pylint/checkers/imports.py

```diff
@@ -446,15 +466,15 @@ def visit_import(self, node: nodes.Import) -> None:
 
             self._add_imported_module(node, imported_module.name)
 
-    @check_messages(*MSGS)
     def visit_importfrom(self, node: nodes.ImportFrom) -> None:
         """Triggered when a from statement is seen."""
         basename = node.modname
         imported_module = self._get_imported_module(node, basename)
+        absolute_name = get_import_name(node, basename)
 
         self._check_import_as_rename(node)
         self._check_misplaced_future(node)
-        self.check_deprecated_module(node, basename)
+        self.check_deprecated_module(node, absolute_name)
```

Comment:
This change is to avoid these failures from tests in #4678:
```
___________________________ test_functional[dot_relative_import] ____________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x113693850>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "dot_relative_import":
E       
E       Unexpected in testdata:
E          7: deprecated-module

pylint/testutils/lint_module_test.py:107: AssertionError
_________________________ test_functional[dot_dot_relative_import] __________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x1138d3cd0>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "dot_dot_relative_import":
E       
E       Unexpected in testdata:
E          7: deprecated-module

pylint/testutils/lint_module_test.py:107: AssertionError
```